File: packages/types/src/TimelineFrameEvent.ts
// Payload for a timeline's per-frame signals (onEnterFrame / onExitFrame / onFrameConstructed):
// the frame being entered and the frame the playhead is leaving (the sentinel -1 before the first
// frame is realized).
export interface TimelineFrameEvent {
  frame: number;
  previousFrame: number;
}

File: crates/flighthq-types/src/timeline_frame_event.rs
//! `TimelineFrameEvent` — the payload carried by per-frame timeline signals.

/// The payload dispatched to `onEnterFrame` / `onExitFrame` / `onFrameConstructed`
/// when a timeline's playhead changes frames.
///
/// `frame` is the 1-based frame being entered; `previous_frame` is the frame the
/// playhead was on before this change (the `-1` sentinel on the first entry, when
/// the timeline has not yet constructed any frame).
#[derive(Copy, Clone, Debug, PartialEq, Eq)]
pub struct TimelineFrameEvent {
    pub frame: u32,
    pub previous_frame: i64,
}
